src/App.tsx

```diff
@@ -1,14 +1,52 @@
-import React from 'react';
+import { useCallback, useMemo, useState } from 'react';
+import { debounce } from 'lodash';
 import './App.scss';
+import { Autocomplete } from './components/Autocomplete';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  delay: number;
+};
+
+export const App: React.FC<Props> = ({ delay = 1000 }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
```

Comment:
Variable `appliedQuery` is not needed. The filtering should be done inside the `Autocomplete` component.